### ISSUE TITLE
Error Analyzing Data File

### ISSUE THREAD

**Issue by BaltimoreBob** [CMRose3355/TomahaqCompanionProgram#2]
I have constructed a method and it runs fine and I can verify the MS3 data by hand, but when I go to analyze the data file in Companion I get an error saying “Error! Object reference not set to an instance of an object.”  I have a complex User Mods table where I have to set the Nterm and K as variable but I believe it is set up alright.  Can someone tell me what this error might be referring to?  I am not sure if it is a bug or something in my method set up, but as I said the method is working I just can't analyze the data.  I tried to paste the UserMods as a file but it didn't seem to work so I just pasted the text below.  It is complicated but as I said it does work...
Thanks!

FALSE | FALSE | TMT0 | 224.1525 | Static |   | K,NPep
-- | -- | -- | -- | -- | -- | --
FALSE | FALSE | TMT2 | 225.1583 | Static |   | K,NPep
FALSE | FALSE | TMT10 | 229.1629 | Static |   | K,NPep
FALSE | FALSE | TMTsh | 235.1768 | Static |   | K,NPep
TRUE | TRUE | CAM | 57.02146 | Static |   | C
FALSE | FALSE | NEM | 125.0477 | Static |   | C
TRUE | TRUE | OX | 15.99491 | Dynamic | * | M
TRUE | TRUE | Phos | 79.96633 | Dynamic | # | S,T,Y
FALSE | FALSE | Acetyl | 42.01056 | Dynamic | @ | NPep
FALSE | FALSE | gg | 114.0429 | Dynamic | * | K
FALSE | FALSE | ggTMT0 | 338.1954 | Dynamic | * | K
FALSE | FALSE | ggTMT2 | 339.2012 | Dynamic | * | K
FALSE | FALSE | ggTMT10 | 343.2058 | Dynamic | * | K
FALSE | FALSE | ggTMTsh | 349.2197 | Dynamic | * | K
FALSE | FALSE | C2N1 | 3.004 | Dynamic | ^ | G
FALSE | FALSE | C3N1 | 4.007 | Dynamic | # | A
FALSE | FALSE | C5N1 | 6.0138 | Dynamic | & | V,P
FALSE | FALSE | C6N1 | 7.0171 | Dynamic | @ | I,L
FALSE | FALSE | C9N1 | 10.0272 | Dynamic | ^ | F,Y
FALSE | FALSE | TMT10OL | 229.1629 | Dynamic | $ | S,T,Y,H
FALSE | FALSE | TMT_Nterm | 229.1629 | Static |   | NPep
TRUE | TRUE | Acetyl | 42.01056 | Dynamic | @ | K
FALSE | TRUE | TMT_K | 229.1629 | Dynamic | $ | K
TRUE | TRUE | Methyl | 14.01565 | Dynamic | & | K
TRUE | TRUE | DiMethyl | 28.0313 | Dynamic | ^ | R
FALSE | FALSE | TMT0NTerm | 224.1525 | Static |   | NPep
TRUE | FALSE | TMT0K | 224.1525 | Dynamic | $ | K
FALSE | TRUE | TMTDynNTerm | 229.1629 | Dynamic | + | NPep
TRUE | FALSE | aTMT0 | 266.1631 | Dynamic | ! | NPep
FALSE | TRUE | aTMT10 | 271.1735 | Dynamic | ! | NPep
TRUE | FALSE | TMT0DnyNTerm | 224.1525 | Dynamic | + | NPep
TRUE | FALSE | pTMT0NTerm | 304.1188 | Dynamic | ~ | NPep
FALSE | TRUE | pTMT10NTerm | 309.1292 | Dynamic | ~ | NPep

**Comment by CMRose3355** [comment]
Hello BaltimoreBob, do you get any other output from the program before the error is output? It looks like there is an object (probably a table) somewhere in the code that isn't being made. If we can figure out which step it failed at I can see if I can fix it.

**Comment by BaltimoreBob** [comment]
Hi Chris,
     Here is the log on the analyze data error.  It  appears to be failing at the extracting MS/MS data step.  I am including my .csv and the UserMods list as well and the method generated.  I had to modify everything since my client wanted modifications on K and Nterminus.  I can send you the .raw files if you would like but that will need an ftp or something.  I did not use the priming run because my colleague said he couldn’t get that to work and this is my first time trying TOMAHAQ.  Hope you had a nice holiday!
Appreciated,
Bob

Robert O’Meally, M.S., Proteomics Specialist
Mass Spectrometry and Proteomics Facility
Johns Hopkins School of Medicine
725 N. Wolfe Street, Room 510 Hunterian Bldg.
Baltimore, MD 21205
Ph: (410) 614-6968/Fax: (410) 614-7566
email:  romeally@jhmi.edu<mailto:romeally@jhmi.edu>

**Comment by CMRose3355** [comment]
Hello Bob, 

Can you please download the new version (1.0.2.0) and before you try the analysis to go the "Advanced Parameters" tab and check the "Print Debugging Messages" box. I have placed some print statements within the code that should help us narrow this down. 

Chris

**Comment by BaltimoreBob** [comment]
Hi Chris,
     I thought I was using the latest version.  We downloaded from Github just two weeks ago.  Which is the latest?
Bob

Robert O’Meally, M.S., Proteomics Specialist
Mass Spectrometry and Proteomics Facility
Johns Hopkins School of Medicine
725 N. Wolfe Street, Room 510 Hunterian Bldg.
Baltimore, MD 21205
Ph: (410) 614-6968/Fax: (410) 614-7566
email:  romeally@jhmi.edu<mailto:romeally@jhmi.edu>

**Comment by BaltimoreBob** [comment]
Oh OK I see the new one now.  Downloading.  I have a standard that I labeled with TMT 0 and then a second set at 1:2:5:10 with TMT 10.  I use this to verify the method is working.  I will run this with the new version later today.  Thanks!
Bob
Ps.  So is there any way to do the analysis on an offline computer?  We are a core lab and when the runs are done we still can’t share the instrument since we have to analyze data on the acquisition computer.  This would be a really nice feature if it does not exist already.  I will let you know when I have rerun with the debugging.

Robert O’Meally, M.S., Proteomics Specialist
Mass Spectrometry and Proteomics Facility
Johns Hopkins School of Medicine
725 N. Wolfe Street, Room 510 Hunterian Bldg.
Baltimore, MD 21205
Ph: (410) 614-6968/Fax: (410) 614-7566
email:  romeally@jhmi.edu<mailto:romeally@jhmi.edu>

**Comment by CMRose3355** [comment]
Yes, you can do the analysis on an offline computer. The automated method creation is the only feature that needs an instrument computer. All other functions should work on a offline computer.

**Comment by BaltimoreBob** [comment]
Oh…I was told by my colleague you had to analyze on the instrument computer.  Maybe he is just an instrument hog!  So with the latest version it fails to create my method.  Here is the error log when I execute.  These are otherwise the same parameters that created my old method.
Thanks!
Bob

--------------------------------------------------------

Robert O’Meally, M.S., Proteomics Specialist
Mass Spectrometry and Proteomics Facility
Johns Hopkins School of Medicine
725 N. Wolfe Street, Room 510 Hunterian Bldg.
Baltimore, MD 21205
Ph: (410) 614-6968/Fax: (410) 614-7566
email:  romeally@jhmi.edu<mailto:romeally@jhmi.edu>

**Comment by CMRose3355** [comment]
I believe that this is related to the unzipping issue that is discussed on the other "issue" raised [here](https://github.com/CMRose3355/TomahaqCompanionProgram/issues/1). 

Try that fix and see if the method can be made.

**Comment by BaltimoreBob** [comment]
Hi Chris,
     That must have been it.  The cloned program made the method.  So where should I see the debugging errors?  I have not run the newest method but it was created successfully.
Bob

Robert O’Meally, M.S., Proteomics Specialist
Mass Spectrometry and Proteomics Facility
Johns Hopkins School of Medicine
725 N. Wolfe Street, Room 510 Hunterian Bldg.
Baltimore, MD 21205
Ph: (410) 614-6968/Fax: (410) 614-7566
email:  romeally@jhmi.edu<mailto:romeally@jhmi.edu>

**Comment by CMRose3355** [comment]
I believe that you mentioned the error was occurring during analysis of data. I would try to analyze the same data that was producing the error previously. I currently only placed messages within the code that does data analysis, and not the code that does method creation.

**Comment by BaltimoreBob** [comment]
Gotcha…here it is…seems like on or after scan 1555 is crashing it?  Where should I look to correct?


Robert O’Meally, M.S., Proteomics Specialist
Mass Spectrometry and Proteomics Facility
Johns Hopkins School of Medicine
725 N. Wolfe Street, Room 510 Hunterian Bldg.
Baltimore, MD 21205
Ph: (410) 614-6968/Fax: (410) 614-7566
email:  romeally@jhmi.edu<mailto:romeally@jhmi.edu>

**Comment by CMRose3355** [comment]
Ok I added in more messages to narrow it down a bit more. It appears that something is missing from the raw file. You can download the version I just published and try the analysis again. 

One thing to look for is to make sure that scan 1555 is an MS2 than has an MS3 associated with it. You can also check the scan header to see if the fields for SPS ions are empty or not. 

What version of Xcalibur and Tune are you running?

**Comment by BaltimoreBob** [comment]
That version didn’t give any new information.  Here is a slicer of the .RAW file.  Scan 1550 is scan 1 on here so 1555 is scan 6.  This appears to be where or shortly before the error is occurring.
Thanks,
Bob

Robert O’Meally, M.S., Proteomics Specialist
Mass Spectrometry and Proteomics Facility
Johns Hopkins School of Medicine
725 N. Wolfe Street, Room 510 Hunterian Bldg.
Baltimore, MD 21205
Ph: (410) 614-6968/Fax: (410) 614-7566
email:  romeally@jhmi.edu<mailto:romeally@jhmi.edu>

**Comment by CMRose3355** [comment]
Hello Bob, 

Please try the newest version again. The last trial helped be narrow in on the issue. There is a small chunk of code that could be the issue and hopefully the newest version will demonstrate where the error is.

**Comment by BaltimoreBob** [comment]
Here you go.  Looks like it gets to 1558 now.

Robert O’Meally, M.S., Proteomics Specialist
Mass Spectrometry and Proteomics Facility
Johns Hopkins School of Medicine
725 N. Wolfe Street, Room 510 Hunterian Bldg.
Baltimore, MD 21205
Ph: (410) 614-6968/Fax: (410) 614-7566
email:  romeally@jhmi.edu<mailto:romeally@jhmi.edu>

**Comment by BaltimoreBob** [comment]
Hi Chris,
     Separate issue.
So I get this error when I try to run Analyze on my desktop PC.  All others attempts were done on the acquisition (Fusion Not Lumos) computer.  Is there another executable I need on my desktop like MSFileReader?  I think I have that but is there something else?  Not Registered?

Robert O’Meally, M.S., Proteomics Specialist
Mass Spectrometry and Proteomics Facility
Johns Hopkins School of Medicine
725 N. Wolfe Street, Room 510 Hunterian Bldg.
Baltimore, MD 21205
Ph: (410) 614-6968/Fax: (410) 614-7566
email:  romeally@jhmi.edu<mailto:romeally@jhmi.edu>

**Comment by CMRose3355** [comment]
Yes, you will need MSFileReader including both 32 and 64 bit versions. I will need to send it via email as I don't know if I can host that here on GitHub.

**Comment by BaltimoreBob** [comment]
Hi Chris,
     Did you get the slicer .Raw file I sent you?  I am noticing that the .Raw file appears to be only picking one ion for SPS.  In the MS3 HCD scans I only see one ion in each scan header.  Should there be at least 4 the way I set my method up?  I assume you are busy so take your time responding.  Just trying to track down where this issue comes from.
Thanks,
Bob

Robert O’Meally, M.S., Proteomics Specialist
Mass Spectrometry and Proteomics Facility
Johns Hopkins School of Medicine
725 N. Wolfe Street, Room 510 Hunterian Bldg.
Baltimore, MD 21205
Ph: (410) 614-6968/Fax: (410) 614-7566
email:  romeally@jhmi.edu<mailto:romeally@jhmi.edu>
